source/base/polynomials_hermite.cc

```diff
@@ -0,0 +1,122 @@
+
+#include <deal.II/base/polynomials_hermite.h>
+
+DEAL_II_NAMESPACE_OPEN
+
+namespace Polynomials
+{
+  namespace
+  {
+    std::vector<double>
+    hermite_poly_coeffs(const unsigned int regularity, const unsigned int index)
+    {
+      AssertIndexRange(index, 2 * regularity + 2);
+
+      const unsigned int curr_index = index % (regularity + 1);
+      const bool         side       = (index > regularity);
+
+        // Protection needed here against underflow errors
+      const int comparison_1 = static_cast<int>(regularity + 1 - curr_index);
+      const int comparison_2 = side ? static_cast<int>(curr_index + 1) :
+                                      static_cast<int>(regularity + 2);
+
+      std::vector<double> poly_coeffs(2 * regularity + 2, 0.0);
+
+      if (side) // g polynomials
+        {
+          int binomial_1 = (curr_index % 2) ? -1 : 1;
+
+          for (int i = 0; i < comparison_2; ++i)
+            {
+              int inv_binomial = 1;
+
+              for (int j = 0; j < comparison_1; ++j)
+                {
+                  int binomial_2 = 1;
+
+                  for (int k = 0; k < j + 1; ++k)
+                    {
+                      poly_coeffs[regularity + i + k + 1] +=
+                        binomial_1 * inv_binomial * binomial_2;
+                      binomial_2 *= k - j;
+                      binomial_2 /= k + 1;
+                    }
+                  inv_binomial *= regularity + j + 1;
+                  inv_binomial /= j + 1;
+                }
+                 // Protection needed here against underflow errors
+              binomial_1 *= -static_cast<int>(curr_index - i);
+              binomial_1 /= i + 1;
+            }
+        }
+      else // f polynomials
+        {
+          int binomial = 1;
+
+          for (int i = 0; i < comparison_2; ++i)
+            {
+              int inv_binomial = 1;
+
+              for (int j = 0; j < comparison_1; ++j)
+                {
+                  poly_coeffs[curr_index + i + j] += binomial * inv_binomial;
+                  inv_binomial *= regularity + j + 1;
+                  inv_binomial /= j + 1;
+                }
+               // Protection needed here against underflow errors
+              binomial *= -static_cast<int>(regularity + 1 - i);
+              binomial /= i + 1;
+            }
+        }
+
+      // rescale coefficients by a factor of 4^curr_index to account for reduced
+      // L2-norms
+      double precond_factor = Utilities::pow(4, curr_index);
+      for (auto &it : poly_coeffs)
+        it *= precond_factor;
+
+      return poly_coeffs;
+    }
+  } // namespace
+
+
+
+  HermitePoly::HermitePoly(const unsigned int regularity,
+                           const unsigned int index)
+    : Polynomial<double>(hermite_poly_coeffs(regularity, index))
+    , degree(2 * regularity + 1)
+    , regularity(regularity)
+    , side_index(index % (regularity + 1))
+    , side((index > regularity) ? 1 : 0)
```

Comment:
This may be easier to read as follows, given the modulus operation in the line above:
```suggestion
    , side((index >= regularity+1) ? 1 : 0)
```